test/unittest/statistics/dds/StatisticsDomainParticipantTests.cpp

```diff
@@ -0,0 +1,436 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <fastdds/dds/core/policy/QosPolicies.hpp>
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/log/Log.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+#include <fastdds/statistics/dds/domain/DomainParticipant.hpp>
+#include <fastdds/statistics/dds/publisher/qos/DataWriterQos.hpp>
+#include <fastdds/statistics/topic_names.hpp>
+#include <fastrtps/types/TypesBase.h>
+#include <statistics/types/typesPubSubTypes.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace statistics {
+namespace dds {
+
+/*
+ * This test checks eprosima::fastdds::statistics::dds::DomainParticipant narrow methods.
+ * 1. Create a eprosima::fastdds::dds::DomainParticipant
+ * 2. Check if the -DFASTDDS_STATISTICS flag has been set and if not check that nullptr is returned even with a valid
+ * DomainParticipant.
+ * 3. Use both narrow methods to obtain the pointer to the children class
+ * 4. Call both narrow methods with an invalid pointer and check that it returns nullptr
+ * 5. Delete DDS entities
+ */
+TEST(StatisticsDomainParticipantTests, NarrowDomainParticipantTest)
+{
+    eprosima::fastdds::dds::DomainParticipant * participant =
+            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->
+            create_participant(0, eprosima::fastdds::dds::PARTICIPANT_QOS_DEFAULT);
+    ASSERT_NE(participant, nullptr);
+
+#ifndef FASTDDS_STATISTICS
+    eprosima::fastdds::statistics::dds::DomainParticipant * statistics_participant =
+            eprosima::fastdds::statistics::dds::DomainParticipant::narrow(participant);
+    EXPECT_EQ(statistics_participant, nullptr);
+
+    ASSERT_TRUE(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant) == 
+            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

Comment:
```suggestion
    ASSERT_EQ(eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant),
            eprosima::fastrtps::types::ReturnCode_t::RETCODE_OK);
```

Reply:
Done in e11835be266a742ca3d2aa570977bd37e720274b